Realdom.php

```diff
@@ -201,7 +201,10 @@
                     next($this->_arguments);
                 }
 
-                while (list($name, $default) = each($this->_arguments)) {
+                do {
```

Comment:
(previous comment lost)
I think you should keep a while() (} loop because $this->_arguments might be already at the end because of the previous next() call

Reply:
From the PHP.net doc, it seems that `each` return the current key / value even, next moves the cursor. The `do {} while ()` loop give the same behavior in our case.

Reply:
The thing is that `while (each() ) {}` doesn't enter into the loop if the pointer is already at the end because of [https://github.com/shulard/Realdom/blob/d16f9feda20552292821c01ba8be9c7d88f9277c/Realdom.php#L201](this line)

So, it could be possible that [https://github.com/shulard/Realdom/blob/d16f9feda20552292821c01ba8be9c7d88f9277c/Realdom.php#L205-L236](all this block) is run while there is no more _arguments.

The `while () {}` prevent this but makes it a little more difficult than the `do {} while();`